### DISCUSSION TITLE
Likert survey

### DISCUSSION THREAD

**Discussion by affective-brain-lab** [jspsych/jsPsych#1589] [Q&A]
Hi,

I create a simple survey and I would like to store the response but I am not sure what to put after share_choice = jsPsych.data.getLastTrialData().values()[0].
Can someone explain what should be put here?

Also, I would like to avoid subjects clicking continue before they indicate their choices, could you help me with that?

Thank you very much!
Below my code so far.

Best
Valentina

```
var share_choice;
var info_share = {
  type: 'survey-likert',
  questions: [
    {prompt: "Would you like to share this information?", labels: scale_1},],
  randomize_question_order: true,
  on_finish: function(data) {
            share_choice = jsPsych.data.getLastTrialData().values()[0].scale_1
  }      
} 
timeline.push(info_share);
```

**Comment by ricardojromeu** [comment]
Hi Valentina,

For your second question, in the "questions" variable, do something like this:

questions: [
        {prompt: , ..., **required: true **}
]
Then they will not be able to move on until they give a response. Here's an example from my own code:
questions:[{prompt: Binary_Questions[i], name: i, labels: Binary_Scale, required: true}]

As for your first question, you could do:
`data.share_choice = jsPsych.data.getLastTrialData().values()[0].responses`
This will give a _string_ that looks like JSON data. You can then parse the string using JSON.parse(**data.share_choice**) to get a JS object.  From there you can get whatever data you need. I highly recommend using console.log() to see piece by piece how jsPsych is interpreting your commands for this, because it can get tricky.

**Reply by becky-gilbert** [reply]
Thanks @ricardojromeu! 
Just wanted to add that this has changed a bit in [jsPsych v6.3.0](https://github.com/jspsych/jsPsych/releases/tag/v6.3.0) (in case @affective-brain-lab or anyone else reading this is using v6.3.0+): the likert responses are stored in a field called `response` (rather than `responses`), and they're stored as an object (rather than a JSON string). So as of v6.3.0, it looks like this if you're using the `name` parameter when setting up the questions:
```js
var share_choice = jsPsych.data.getLastTrialData().values()[0].response.my_question_name;
```
Or like this if you're using the default question names of `Q0`, `Q1`, etc.
```js
var share_choice = jsPsych.data.getLastTrialData().values()[0].response.Q0;
```
